From 247a48bf73eaa129aafa90a9475688d766abbe34 Mon Sep 17 00:00:00 2001
From: tx0 <me@tx0.su>
Date: Tue, 29 Apr 2025 12:12:25 +0000
Subject: [PATCH] Revert "linux: up-device-supply-battery: reset the status to
 discharging when current_now is negative"

This reverts commit d7ff457784eca4de01673012506cf6ac027849a3.
---
 src/linux/up-device-supply-battery.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/linux/up-device-supply-battery.c b/src/linux/up-device-supply-battery.c
index 9bca318..8ede873 100644
--- a/src/linux/up-device-supply-battery.c
+++ b/src/linux/up-device-supply-battery.c
@@ -306,17 +306,8 @@ up_device_supply_battery_refresh (UpDevice *device,
 		values.percentage = g_udev_device_get_sysfs_attr_as_double_uncached (native, "capacity");
 		values.percentage = CLAMP(values.percentage, 0.0f, 100.0f);
 	}
-
-	/* For some battery solutions, for example axp20x-battery, the kernel reports
-	 * status = charging but the battery actually discharges when connecting a
-	 * charger. Upower reports the battery is "discharging" when current_now is
-	 * found and is a negative value as long as the battery isn't fully charged.*/
 	values.state = up_device_supply_get_state (native);
 
-	if (values.state != UP_DEVICE_STATE_FULLY_CHARGED &&
-	    g_udev_device_get_sysfs_attr_as_double_uncached (native, "current_now") < 0.0)
-		values.state = UP_DEVICE_STATE_DISCHARGING;
-
 	values.temperature = g_udev_device_get_sysfs_attr_as_double_uncached (native, "temp") / 10.0;
 
 	up_device_battery_report (battery, &values, reason);
-- 
2.49.0

